src/Control/SearchEditor.php

```diff
@@ -173,19 +173,38 @@ protected function applyChanges(Filter\Rule $rule, array &$values, array $path =
 
             $newOperator = $this->popKey($values, $identifier . '-operator');
             if ($newOperator !== null && QueryString::getRuleSymbol($rule) !== $newOperator) {
+                $value = $rule->getValue();
+                $column = $rule->getColumn();
                 switch ($newOperator) {
                     case '=':
-                        return Filter::equal($rule->getColumn(), $rule->getValue());
+                        if (is_string($value) && strpos($value, '*') !== false) {
+                            return Filter::similar($column, $value);
+                        }
+
+                        return Filter::equal($column, $value);
                     case '!=':
-                        return Filter::unequal($rule->getColumn(), $rule->getValue());
+                        if (is_string($value) && strpos($value, '*') !== false) {
+                            return Filter::unlike($column, $value);
+                        }
+
+                        return Filter::unequal($column, $value);
                     case '>':
-                        return Filter::greaterThan($rule->getColumn(), $rule->getValue());
+                        return Filter::greaterThan($column, $value);
                     case '>=':
-                        return Filter::greaterThanOrEqual($rule->getColumn(), $rule->getValue());
+                        return Filter::greaterThanOrEqual($column, $value);
                     case '<':
-                        return Filter::lessThan($rule->getColumn(), $rule->getValue());
+                        return Filter::lessThan($column, $value);
                     case '<=':
-                        return Filter::lessThanOrEqual($rule->getColumn(), $rule->getValue());
+                        return Filter::lessThanOrEqual($column, $value);
+                }
+            }
+
+            $value = $rule->getValue();
+            if (is_string($value) && strpos($value, '*') !== false) {
+                if (QueryString::getRuleSymbol($rule) === '=') {
+                    return Filter::similar($rule->getColumn(), $value);
+                } elseif (QueryString::getRuleSymbol($rule) === '!=') {
+                    return Filter::unlike($rule->getColumn(), $value);
                 }
             }
```

Comment:
At first I didn't knew what this section is for. After thinking about it, I do: Because the value can also change from non-wildcard to wildcard. Though, it would make more sense if you'd place this section inside the structure at line no. 170. It makes the intention of the code more clear, and it doesn't run needlessly. Though, you can't just return a new rule there, the operator might still change after all. So maybe don't move it, but also check here if the value has changed and don't replace the rule needlessly.

Reply:
Did not move this block but added a condition, so that the rule changes only when the value has been changed.